data-prepper-plugins/avro-codecs/src/main/java/org/opensearch/dataprepper/avro/AbstractAvroEventConverterTemplate.java

```diff
@@ -0,0 +1,70 @@
+package org.opensearch.dataprepper.avro;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.opensearch.dataprepper.model.sink.OutputCodecContext;
+
+import java.util.List;
+import java.util.Map;
+
+abstract class AbstractAvroEventConverterTemplate implements AvroEventConverter {
+    private final SchemaChooser schemaChooser;
+
+    protected AbstractAvroEventConverterTemplate(final SchemaChooser schemaChooser) {
+        this.schemaChooser = schemaChooser;
+    }
+
+    @Override
+    public GenericRecord convertEventDataToAvro(final Schema schema,
+                                                final Map<String, Object> eventData,
+                                                final OutputCodecContext codecContext) {
+        return convertEventDataToAvro(schema, eventData, codecContext, true);
+    }
+
+    private GenericRecord convertEventDataToAvro(final Schema schema,
+                                                 final Map<String, Object> eventData,
+                                                 final OutputCodecContext codecContext,
+                                                 boolean rootOfData) {
+        final GenericRecord avroRecord = new GenericData.Record(schema);
+
+        for (String key : getKeyNames(schema, eventData, codecContext, rootOfData)) {
+            final Schema.Field field = schema.getField(key);
+            if (field == null) {
+                throw new RuntimeException("The event has a key ('" + key + "') which is not included in the schema.");
```

Comment:
What is the behavior for this exception? We include it as one of the Events that has to be dropped?

Reply:
This will show in the sampled log as merged in #3242.